src/launchpad/size/symbols/swift_aggregator.py

```diff
@@ -0,0 +1,94 @@
+"""Aggregator for Swift symbols by module/type."""
+
+from __future__ import annotations
+
+from typing import NamedTuple
+
+import sentry_sdk
+
+from launchpad.size.symbols.macho_symbol_sizes import SymbolSize
+from launchpad.size.symbols.types import SwiftSymbolList, SwiftSymbolTypeGroup
+from launchpad.utils.apple.cwl_demangle import CwlDemangler
+from launchpad.utils.logging import get_logger
+
+logger = get_logger(__name__)
+
+
+class SwiftModuleType(NamedTuple):
+    module: str
+    type_name: str
+
+
+class SwiftSymbolTypeAggregator:
+    """Aggregates symbols by their module/type after demangling."""
+
+    def __init__(self) -> None:
+        self.demangler = CwlDemangler()
+
+    @staticmethod
+    def is_swift_symbol(mangled_name: str) -> bool:
+        """Check if a symbol is a Swift symbol based on its mangled name.
+
+        Includes:
+        - _$s: Modern Swift mangling
+        - _Tt: Older Swift mangling
+        - __IVARS__Tt: ObjC runtime metadata for Swift classes
+        - __DATA__Tt: Data section symbols for Swift types
+        """
+        if mangled_name.startswith("_$s") or mangled_name.startswith("_Tt"):
+            return True
+        # Swift classes exposed to ObjC have metadata with Swift mangling
+        if "__Tt" in mangled_name or "__TtC" in mangled_name:
```

Comment:
isn't `"__TtC" in mangled_name` redundant since `"__Tt" in mangled_name` would always be true in that case too?